### PR TITLE
Updated the example of ssr-caching to utilize the App Router.

### PR DESCRIPTION
**Pull request by PapatMayuri**
This PR updates the ssr-caching example for using the App Router. Here are the changes that have been made:

- Renamed the "pages" folder to the "app" folder.
- Added the layout.tsx file as part of the App Router.
- Updated the package.json file.

CC: @samcx

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 918826f9293e02cc554ce14c13e0fb97a7177ea0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/ssr-caching/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const revalidate = 10;
```
Comment (ijjk) [inline examples/ssr-caching/app/page.tsx:1]: This is changing the behavior of what the example was originally showing. Originally the example was showing caching at the CDN layer via `Cache-Control` headers alone, using `export const revalidate` opts the page into ISR which has build time level caching and other features to it. 

We can still convert this to app router although it requires making this page dynamic via `export const dynamic = 'force-dynamic'` and then setting the `Cache-Control` header in `next.config` or `middleware`.

**Review by ijjk — DISMISSED** [review]
See above comment for changes

File: examples/ssr-caching/app/page.tsx (hunk 1 above)
Comment (samcx) [reply examples/ssr-caching/app/page.tsx:1]: @PapatMayuri Actually to clarify, neither `next.config` and Next.js Middleware work for overriding the `Cache-Control` header in App Router. We'll have to wait for one of those options to be fixed before proceeding with this PR.

Comment (ijjk) [reply examples/ssr-caching/app/page.tsx:1]: We can land this now that https://github.com/vercel/next.js/pull/69802 has landed

Comment (PapatMayuri) [reply examples/ssr-caching/app/page.tsx:1]: Thanks, @ijjk I will Update this PR tomorrow Morning.

Comment (PapatMayuri) [reply examples/ssr-caching/app/page.tsx]: @samcx I have updated PR as per the comment please check.

**Review by samcx — APPROVED** [review]

**Pull request merged**